phoenix-ios/phoenix-ios/officers/BusinessManager.swift

```diff
@@ -153,14 +153,21 @@ class BusinessManager {
 			.store(in: &cancellables)
 		
 		// In-flight payments observer
-		business.paymentsManager.inFlightOutgoingPaymentsPublisher()
-			.sink { (count: Int) in
+		business.peerManager.peerStatePublisher()
+			.flatMap { $0.eventsFlowPublisher() }
+			.sink { (event: Lightning_kmpPeerEvent) in
 				
-				log.debug("inFlightOutgoingPaymentsPublisher: count = \(count)")
-				if count > 0 {
-					self.beginLongLivedTask()
-				} else {
-					self.endLongLivedTask()
+				if let paymentProgress = event as? Lightning_kmpPaymentProgress {
+					let paymentId = paymentProgress.request.paymentId.description()
+					self.beginLongLivedTask(id: paymentId)
+					
+				} else if let paymentSent = event as? Lightning_kmpPaymentSent {
+					let paymentId = paymentSent.request.paymentId.description()
+					self.endLongLivedTask(id: paymentId)
+					
+				} else if let paymentNotSent = event as? Lightning_kmpPaymentNotSent {
+					let paymentId = paymentNotSent.request.paymentId.description()
+					self.endLongLivedTask(id: paymentId)
```

Comment:
If I understand https://github.com/ACINQ/lightning-kmp/pull/550 correctly, `SensitiveTaskEvents` are only published for splicing updates. That's why on Phoenix's side we also need to monitor outgoing payment events. If we did not do that, we would not trigger long lived tasks for when sending LN payments, which would be a regression.

If that's correct, we should add a comment saying that in the long run, `SensitiveTaskEvents` will also cover the LN payment use case and then we'll be able to remove this custom section.

Is that correct @pm47 @robbiehanson ?

Reply:
Yes that's correct, it's only a proof of concept for now, it also has the advantage to cover a case that couldn't be handled from the outside of lightning-kmp. We can later extend it to other tasks like payments.